### ISSUE TITLE
@nx/plugin:generator error in Nx 20

### ISSUE THREAD

**Issue by Tungsten78** [nrwl/nx#29127]
### Current Behavior

Following the [docs](https://nx.dev/extending-nx/recipes/local-generators#creating-a-generator) an error is produced when issuing `nx generate @nx/plugin:generator tools/my-plugin/src/generators/my-generator`.

```
 NX   C:\Users\Chris Leigh\Source\repos\nx-vigil\node_modules\@nx\plugin\src\generators\generator\files\generator\generator.spec.ts.template:7

    5| import { <%= schemaInterfaceName %> } from './schema';
    6|
 >> 7| describe('<%= name %> generator', () => {
    8|   let tree: Tree;
    9|   const options: <%= schemaInterfaceName %> = { name: 'test' };
    10|

name is not defined
```

### Expected Behavior

Generator creates files.

WORKAROUND:

If the [optional name argument](https://github.com/nrwl/nx/blob/ec5a5e63606b4b7a797ef1ff9d211ffa47dcef0c/packages/plugin/src/generators/generator/schema.json#L15) is provided, the generator works.

### Steps to Reproduce

1.
Follow steps from [docs](https://nx.dev/extending-nx/recipes/local-generators#creating-a-generator) 

```
nx add @nx/plugin
nx g @nx/plugin:plugin tools/my-plugin
```

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node           : 22.9.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 10.5.0

nx (global)        : 20.0.6
nx                 : 20.1.4
@nx/js             : 20.1.4
@nx/jest           : 20.1.4
@nx/eslint         : 20.1.4
@nx/workspace      : 20.1.4
@nx/angular        : 20.1.4
@nx/cypress        : 20.1.4
@nx/devkit         : 20.1.4
@nx/eslint-plugin  : 20.1.4
@nx/plugin         : 20.1.4
@nx/web            : 20.1.4
@nx/webpack        : 20.1.4
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx-dotnet/core
@nx-tools/nx-container
---------------------------------------
Community plugins:
@auth0/auth0-angular         : 2.2.1
@ngrx/component              : 17.0.1
@ngrx/component-store        : 17.0.1
@nx-dotnet/core              : 2.1.1
@nx-tools/nx-container       : 5.1.0
@nxlv/python                 : 17.0.0
@trumbitta/nx-plugin-openapi : 1.12.1
---------------------------------------
Local workspace plugins:
         core
```

### Failure Logs

```shell

```

### Operating System

- [ ] macOS
- [ ] Linux
- [x] Windows
- [ ] Other (Please specify)

**Comment by Oneia** [comment]
have you figure out something?

**Comment by leosvelperez** [comment]
This was fixed by https://github.com/nrwl/nx/pull/28724. The fix will be released soon as part of Nx v20.2.0.
